Gateway/Response/PaymentRefundDetailsHandler.php

```diff
@@ -39,6 +39,10 @@ public function handle(array $handlingSubject, array $response)
         $payment = $payment->getPayment();
 
         foreach ($response as $singleResponse) {
+            if (isset($singleResponse['error'])) {
+                throw new \Magento\Framework\Exception\LocalizedException(__($singleResponse['error']));
```

Comment:
If you can confirm that this error is already logged in a file I think it is better to show a generic error message `The refund failed, check the logs for details.`, that way your concern about the 'Field amount' error is also solved.

Reply:
Updated this